addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1258,6 +1262,94 @@ options.registry.parallax = options.Class.extend({
     onMove: function () {
         this._refreshPublicWidgets();
     },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Build/remove parallax.
+     *
+     * @see this.selectClass for parameters
+     */
+    parallax: function (previewMode, widgetValue, params) {
+        const isParallax = widgetValue === 'true';
+        this.$target.toggleClass('parallax', isParallax);
+        if (isParallax) {
+            this._buildParallax();
+        } else {
+            this._removeParallax();
+        }
+        if (!previewMode) {
+            this.isParallax = isParallax ? true : false;
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeVisibility: function () {
+        let $el = this.$target.find('> .s_parallax_bg');
+        $el = $el[0] ? $el : this.$target;
+        const hasBackgroundImage = $el.css('background-image') !== 'none';
+        if (!hasBackgroundImage) {
+            this._removeParallax();
+            this.$target.removeClass('oe_img_bg');
+        }
+        const showParallaxOptions = hasBackgroundImage && !this.$target.hasClass('o_background_video');
+        return this._super(...arguments) && showParallaxOptions;
+    },
+    /**
+     * @private
+     */
+    _buildParallax: function () {
+        if (!this.$target.find('> .s_parallax_bg').length) {
+            $('<span/>', {
+                class: 's_parallax_bg oe_img_bg ',
+                style: 'background-position: 50% 0;',
+            }).prependTo(this.$target);
+            this._updateBackgroundOptions();
+            // FIXME big hack, this should be improved
+            if (!this.isParallax) {
```

Comment:
Your `_buildParallax` function is always called under the condition of `this.isParallax` so this condition here can never be true. So likely that this code is not needed at all? My guess is that the `isParallax` value is also useless then. Could you review? Thanks.

Reply:
The condition for _buildParallax is widgetValue = true.
So if the value is true we build the parallax. And then the "if (!this.isParallax)" is usefull to said if it was already in parallax before or not. If it was not already in parallax we change the colorpicker value to black-50. I agree that it's strange that we have to rebuilt the parallax if it was already in parallax. But it is the "preview" which removes it at each mode change. Maybe I did something wrong but  we need it like thaht to make it work :)

Reply:
Actually, it is simpler than that then: there is not supposed to have a preview mode for parallax. Normally not the case since 12.0/13.0? And it should stay like that :+1:

Reply:
preview mode works in master. but ok I will remove it +1